test/__init__.py

```diff
@@ -205,13 +205,13 @@ def random_basic_string(length=16, *, prefix=None, digit_spacing=None):
         prefix = random.choice(string.ascii_lowercase)
     random_length = length - len(prefix)
     if digit_spacing is None:
-        chars = [random.choice(string.ascii_lowercase + string.digits) for _ in range(random_length)]
+        chars = "".join([random.choice(string.ascii_lowercase + string.digits) for _ in range(random_length)])
```

Comment:
Could make do without the extra list here (`chars = "".join(random.choice(string.ascii_lowercase + string.digits) for _ in range(random_length))`).